### PR TITLE
ENT-9366: Added naming conventions / guidelines for CFEngine Build modules

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Review by larsewi — APPROVED** [review]
ACK :heavy_check_mark:

**Review by craigcomstock — APPROVED** [review]
🎉 🌮  :)

**Review by nickanderson — COMMENTED** [review]
Some questions.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -23,6 +23,24 @@ In order to submit a module to CFEngine build, you need:
...
+
+In order for other users to more easily find your module and understand what it does, we have some guidelines on how to name it:
+
+* Module names are lowercase, with dashes (`-`) to separate words
+* Some prefixes are commonly used:
+  * `cve-...` - (followed by ID), modules used to address CVEs (either by making changes, or just identifying vulnerable hosts) 
+  * `inventory-...` - modules which add inventory (reporting) data and don't make significant changes to the system
+  * `promise-type-...` - modules which provide a promise type
+  * `library-...` - modules which don't do much on their own, but are normally used as dependencies for other modules
+  * `compliance-report-...` - modules which provide a compliance report in mission portal
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply CONTRIBUTING.md:36]: The answer is, as always, it depends :)

I think we have to try to find out what makes sense on a case by case basis. In general putting both inventory and compliance report in the same module is nice, I like to just add one module, not 1 module + 1 dependency every time. In that case, I would name the module `compliance-report-...`. When those inventory attributes seem very useful and valuable to have by themself, or if you want to reuse them across multiple modules, it makes sense to make a separate module.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -23,6 +23,24 @@ In order to submit a module to CFEngine build, you need:
...
+  * `example-...` - modules which will always be just an example, never intended to become a real / supported module
+  * `ssh-...` - modules relating to SSH
+  * `sudo-...` - modules relating to sudo
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply CONTRIBUTING.md:39]: It depends :)

I don't think modules should always start with a package name, but in cases like this, where we have many modules relating to the same "thing" (such as ssh or sudo or another package), I think it's a good idea that they all have similar names.

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (olehermanse) [reply CONTRIBUTING.md:36]: I can try, but I don't really know how much it helps: In the end we review all modules submitted as PRs and can suggest / require naming changes before acceptaing them into the index.

[the remark this answers is not in the document]
Comment (olehermanse) [reply CONTRIBUTING.md:36]: https://github.com/cfengine/build-index/pull/284

**Pull request merged**